### PR TITLE
barista(error-pages): Add error pages to barista, wip

### LINKED ISSUE #108: Barista: Error pages
create error pages for 404,...

### PR DESCRIPTION
**Pull request by NinaKammerlander**
### <strong>Pull Request</strong>
closes #108

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: apps/barista/src/pages/error-page/error-page.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  templateUrl: 'error-page.html',
+  styleUrls: ['error-page.scss'],
+  host: {
+    role: 'main',
+  },
+})
+export class BaErrorPage implements BaPage {
+  contents: BaErrorPageContent;
+
+  constructor() {}
```
Comment (lara-aigmueller) [inline apps/barista/src/pages/error-page/error-page.ts]: Can we remove this one?

File: libs/shared/barista-definitions/src/lib/barista-definitions.ts [hunk 2]
```diff
@@ -51,6 +52,11 @@ export interface BaPageMetaBase {
   navGroup?: string;
 }
 
+export interface BaErrorPageContent extends BaPageMetaBase {
+  layout: BaLayoutType.Error;
+  errormessage: string;
```
Comment (thomaspink) [inline libs/shared/barista-definitions/src/lib/barista-definitions.ts]: Can we change this to `content` to be in line with the other types

File: apps/barista/src/shared/page.service.ts [hunk 3]
```diff
@@ -14,18 +14,36 @@
...
+  layout: BaLayoutType.Error,
+  errormessage:
+    'Sorry, the page you tried to access does not exist. Are you using an outdated link?',
+};
+
+const ERRORPAGE: BaErrorPageContent = {
+  title: 'Oops!',
+  layout: BaLayoutType.Error,
+  errormessage:
+    "Sorry, an error has occured. Don't worry, we're working to fix the problem!",
```
Comment (thomaspink) [inline apps/barista/src/shared/page.service.ts]: Should we change the error message or implement a way so we get notified that this problem has been encountered (How can we fix a problem we do not know about ;) )

File: apps/barista/src/shared/page.service.ts [hunk 4]
```diff
@@ -71,11 +89,16 @@ export class BaPageService {
 
     this.http
       .get<BaSinglePageContent>(requestPath, { responseType: 'json' })
-      .pipe
-      // tap(data => {
-      //   console.log(data);
-      // })
-      ()
+      .pipe(
+        tap(data => {
```
Comment (thomaspink) [inline apps/barista/src/shared/page.service.ts]: Pls change it to a map, and directly return `ERRORPAGE` if the data is invalid.

**Comment by lukasholzer** [comment]
Please rebase current master inside in case you have an outdated circleci file.

**Pull request merged**